src/store.ts

```diff
@@ -178,6 +178,20 @@ export class Store {
     }
   }
 
+  invalidate(
+    rootKey: string,
```

Comment:
Couldn't this just accept the output from resolve?

Reply:
The output from resolve is `Author:1` or `['Author:1', 'Author:2']`. We don't know where these come from just yet

Reply:
But isn't that the intent? We don't want to just invalidate specific links but most likely also be able to invalidate entities?

I find it more likely for a user to invalidate "all todos" or "all users" for instance than the field that points at those from queries

Reply:
I've been playing with this but it seems rather dangerous to me: we allow resolve and invalidate by means of the `dependencies` Set. To invalidate everything of a certain kind through links is the easiest way else we'd need to make assumptions as for example:
We get `Author:1` then we can assume that we do removeRecord(Author:1.id | _id) and removeRecord(Author:1.__typename)

Reply:
It depends, I'd say. If we ignore the case of ignoring a set of entities with the same typename at least, since we're missing an API for that, we still have to look at the input for invalidations.

Invalidating links is essentially sometimes OK, but most of the time it'll lead to confusion. The user invalidates what they think is a normalised piece of data but will only see a specific field on a specific query path disappear.

That's potentially where we'd get into a situation where Apollo is of unpredictability.

Reply:
I see what you mean, don't really have a clearcut answer for you though. It's a complex scenario